File: python/generate_random_number.py
'''
generate random numbers following normal distribution.
'''
import random
import math


def normal_random():
    """
    :return: a normally distributed random number
    """
    y_1 = y_2 = random.random()

    return math.cos(2 * 3.14 * y_2) * math.sqrt(-2. * math.log(y_1))


while True:
    medium = input("Input medium value: ")
    variance = input("Input variance: ")
    number = input("Input the number of random numbers: ")

    for day in range(int(number)):
        result = normal_random() * float(variance) + float(medium)
        print(result)
        # 保留三位有效数字
        print("{:.3g}".format(result))

File: c-cpp/generate_random_number.cpp
#include <cstdlib>
#include <cmath>
#include <ctime>
#include <iostream>

using namespace std;

// return a uniformly distributed random number
double RandomGenerator()
{
  return ((double)(rand()) + 1.) / ((double)(RAND_MAX) + 1.);
}

// return a normally distributed random number
double NormalRandom()
{
  double y1 = RandomGenerator();
  double y2 = RandomGenerator();

  return cos(2 * 3.14 * y2) * sqrt(-2. * log(y1));
}

int main()
{
  int number;
  double variance, medium;

  while (1) {
    cout << endl << "Input medium value, variance and the number of random numbers: " << endl;
    cin >> medium >> variance >> number;

    for (int i = 0; i < number; i++) {
      double x = NormalRandom() * variance + medium;
      cout << x << endl;
    }
  }

  return 0;
}
